metron-platform/metron-parsing/metron-parsers/src/main/java/org/apache/metron/parsers/leef/LEEFParser.java

```diff
@@ -0,0 +1,236 @@
+ */
+package org.apache.metron.parsers.leef;
+
+import java.lang.invoke.MethodHandles;
+import java.nio.charset.Charset;
+import java.text.SimpleDateFormat;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.metron.parsers.BasicParser;
+import org.apache.metron.parsers.ParseException;
+import org.apache.metron.parsers.cef.CEFParser;
+import org.apache.metron.parsers.utils.DateUtils;
+import org.apache.metron.parsers.utils.SyslogUtils;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LEEFParser extends BasicParser {
+	private static final long serialVersionUID = 1L;
+
+	protected static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+	private static final String HEADER_CAPTURE_PATTERN = "[^\\|]*";
+	private static final Charset UTF_8 = Charset.forName("UTF-8");
+	private static final String EXTENSION_CAPTURE_PATTERN = "(?<!\\\\)=";
+
+	private Pattern p;
+	private Pattern pext;
+
+	public void init() {
+
+		// CEF Headers: Device Vendor|Device Product|Device Version|Device Event
+		// Class ID|Name|Severity
+
+		String syslogTime = "(?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:tember)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?)\\b +(?:(?:0[1-9])|(?:[12][0-9])|(?:3[01])|[1-9]) (?!<[0-9])(?:2[0123]|[01]?[0-9]):(?:[0-5][0-9])(?::(?:(?:[0-5]?[0-9]|60)(?:[:.,][0-9]+)?))(?![0-9])?";
+		String syslogTime5424 = "(?:\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(?:\\.\\d+)?(?:Z|[+-]\\d{2}:\\d{2}))";
+		String syslogPriority = "<(?:[0-9]+)>";
+		String syslogHost = "[a-z0-9\\.\\\\-_]+";
+
+		StringBuilder sb = new StringBuilder("");
+		sb.append("(?<syslogPriority>");
+		sb.append(syslogPriority);
+		sb.append(")?");
+		sb.append("(?<syslogTime>");
+		sb.append(syslogTime);
+		sb.append("|");
+		sb.append(syslogTime5424);
+		sb.append(")?");
+
+		sb.append("(?<syslogHost>");
+		sb.append(syslogHost);
+		sb.append(")?");
+
+		sb.append(".*");
+
+		sb.append("LEEF:(?<Version>1.0|2.0|0)?\\|");
+
+		headerBlock("DeviceVendor", sb);
+		sb.append("\\|");
+		headerBlock("DeviceProduct", sb);
+		sb.append("\\|");
+		headerBlock("DeviceVersion", sb);
+		sb.append("\\|");
+		headerBlock("DeviceEvent", sb);
+		sb.append("\\|");
+		
+		// add optional delimiter header (only applicable for LEEF 2.0)
+		sb.append("(");
+		headerBlock("Delimiter", sb);
+		sb.append("\\|");
+		sb.append(")?");
+		
+		// extension capture:
+		sb.append(" ?(?<extensions>.*)");
+		String pattern = sb.toString();
+
+		p = Pattern.compile(pattern);
+
+		pext = Pattern.compile(EXTENSION_CAPTURE_PATTERN);
+	}
+
+	@SuppressWarnings("unchecked")
+	public List<JSONObject> parse(byte[] rawMessage) {
+		List<JSONObject> messages = new ArrayList<>();
+
+		String cefString = new String(rawMessage, UTF_8);
```

Comment:
Just making a note as this relates to another discussion on default charsets - https://github.com/apache/metron/pull/1341#issuecomment-493459303. Regardless of which comes first, we should land on a cohesive strategy for all parsers and be sure this parser has the change once both PRs are complete.